src/core/tests/model.cpp

```diff
@@ -1932,3 +1932,21 @@ TEST(model, clone_model) {
     const auto res = fc.compare(model, cloned_model);
     EXPECT_TRUE(res.valid) << res.message;
 }
+
+TEST(model, set_meta_information) {
+    auto arg0 = std::make_shared<ov::opset8::Parameter>(ov::element::f32, ov::PartialShape{1});
+    arg0->set_friendly_name("data");
+    arg0->get_output_tensor(0).set_names({"input"});
+
+    auto relu = std::make_shared<ov::opset8::Relu>(arg0);
+    relu->set_friendly_name("relu");
+    relu->get_output_tensor(0).set_names({"relu_t", "identity"});
+    auto f = std::make_shared<ov::Model>(relu, ov::ParameterVector{arg0});
+
+    std::string key = "data";
+    EXPECT_FALSE(f->has_rt_info<std::string>(key, "test"));
+    EXPECT_THROW(f->get_rt_info<std::string>(key, "test"), ov::Exception);
+    f->set_rt_info("test", key, "test");
+    EXPECT_TRUE(f->has_rt_info<std::string>(key, "test"));
+    EXPECT_NO_THROW(f->get_rt_info<std::string>(key, "test"));
```

Comment:
can we also check that if we use different template types, it returns `false` or exception?